### PR TITLE
fix not existed relative path.

### PR DESCRIPTION
**Pull request by vwxyzh**
@chenkennt @ansyral @hellosnow @superyyrrzz @qinezh

### CODE REVIEW

**Comment by superyyrrzz** [comment]
:shipit:

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs [hunk 1]
```diff
@@ -376,6 +376,7 @@ private static void UpdateHref(HtmlAgilityPack.HtmlNode link, string attribute,
                     Logger.LogWarning($"File {path} is not found in {relativePath}.");
                     // TODO: what to do if file path not exists?
                     // CURRENT: fallback to the original one
+                    path = (TypeForwardedToRelativePath)path - (TypeForwardedToRelativePath)relativePath;
```
Comment (ansyral) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: >path [](start = 20, length = 4)

seems that the anchor would appended twice?

**Comment by qinezh** [comment]
:shipit:

File: src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs [hunk 2]
```diff
@@ -353,47 +353,36 @@ private static void UpdateXref(HtmlAgilityPack.HtmlNode node, IDocumentBuildCont
...
+                    if (path != null)
+                    {
+                        href = (path.UrlDecode().RemoveWorkingFolder() - (TypeForwardedToRelativePath)relativePath).UrlEncode();
+                    }
+                    else
                     {
-                        path += anchor;
-                        link.Attributes.Remove("anchor");
+                        href = originalHref;
                     }
```
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Build.Engine/TemplateModelTransformer.cs]: can remove

**Pull request merged**